cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -1926,7 +1934,7 @@ impl ASTNode<Option<cst::Literal>> {
             cst::Literal::Num(n) => match i64::try_from(*n) {
                 Ok(i) => Some(ExprOrSpecial::Expr(construct_expr_num(i, src.clone()))),
                 Err(_) => {
-                    errs.push(ParseError::ToAST(format!("Literal {n} is too large")));
+                    errs.push(ParseError::NumericLiteralOutOfRange(n.to_string()));
```

Comment:
I wonder if we can let the variant `NumericLiteralOutOfRange` to carry the number instead of the error message? And use `#[error(...)]` to specify the message.